lib/obs.rb

```diff
@@ -0,0 +1,148 @@
+require 'faraday'
+
+# HTTP OBS client for searches
+module OBS
+  # Error class for unsupported search terms
+  class InvalidSearchTerm < StandardError; end
+
+  # One binary that represents an rpm.
+  # It has the following properties
+  # Binary.arch => [String]
+  # Binary.baseproject => [String]
+  # Binary.filename => [String]
+  # Binary.filepath => [String]
+  # Binary.name => [String]
+  # Binary.package => [String]
+  # Binary.project => [String]
+  # Binary.release => [String]
+  # Binary.repository => [String]
+  # Binary.type => [String]
+  # Binary.version => [String]
+  class Binary < Hashie::Mash
+    include Hashie::Extensions::Mash::SymbolizeKeys
+    def self.coerce(binary)
+      if binary.is_a?(Array)
+        binary.map { |bin| Binary.new(bin) }
+      else
+        [Binary.new(binary)]
+      end
+    end
+  end
+
+  # A collection of one more binaries with the following properties
+  # Collection.binaries => [Array] of [Binary]
+  # Collection.matches  => [Integer]
+  class Collection < Hashie::Mash
+    include Hashie::Extensions::Mash::SymbolizeKeys
+    include Hashie::Extensions::Coercion
+    coerce_key :binary, Binary
+    coerce_key :matches, Integer
+
+    def self.coerce(hash)
+      c = Collection.new(hash)
+      c[:binaries] = c.delete(:binary)
+      c
+    end
+  end
+
+  # Helper class that turns a "collection" attribute in HTTP responses into an
+  # instance of [Collection]
+  class Response < Hashie::Mash
+    include Hashie::Extensions::Mash::SymbolizeKeys
+    include Hashie::Extensions::Coercion
+    coerce_key :collection, Collection
+  end
+
+  # Utility function to generate a xpath query from a search term and the following options:
+  # @param [Hash] opts the options for the query.
+  # @option opts [String] :baseproject OBS base project used to search packages for
+  # @option opts [String] :project OBS specific project used to search packages into
+  # @option opts [Boolean] :exclude_debug Exclude Debug packages from search
+  # @option opts [String] :exclude_filter Exclude packages containing this term
+  #
+  def self.xpath_for(query, opts = {})
+    baseproject = opts[:baseproject]
+    project = opts[:project]
+    exclude_debug = opts[:exclude_debug]
+    exclude_filter = opts[:exclude_filter]
+
+    words = query.split(" ").reject { |part| part.match(/^[0-9_\.-]+$/) }
+    versrel = query.split(" ").select { |part| part.match(/^[0-9_\.-]+$/) }
```

Comment:
same here. And you may want o assign this match to a method (repeated code) :wink: